src/parsing/parser.ts

```diff
@@ -207,88 +208,60 @@ export function parseRuleSyntax(
 export function parseMappedTrackerSyntax(
   syntax: MappedTrackerJSON
 ): MappedTrackerDefinition {
-  let keyType = syntax.keyType.trim();
-  let valueType = syntax.valueType.trim();
-  var trackerInitialKeys: any[] = [];
-  var trackerInitialValues: any[] = [];
-  for (var pair of syntax.initialvalues) {
+  let keyType = syntax.keyType;
+  let valueType = syntax.valueType;
+  var keys = syntax.initialvalues.map((val) => val.key);
+  var values = syntax.initialvalues.map((val) => val.value);
+  var trackerInitialKeys: any[] = encodeTrackerData(keys, keyType);
+  var trackerInitialValues: any[] = encodeTrackerData(values, valueType);
+
+  const keyTypeEnum = (PT.find((_pt) => (_pt.name = keyType)) ?? PT[4])
+    .enumeration;
+  const valueTypeEnum = (PT.find((_pt) => (_pt.name = valueType)) ?? PT[4])
+    .enumeration;
+
+  return {
+    name: syntax.name,
+    keyType: keyTypeEnum,
+    valueType: valueTypeEnum,
+    initialKeys: trackerInitialKeys,
+    initialValues: trackerInitialValues,
+  };
+}
+
+function encodeTrackerData(valueSet: any[], keyType: string): any[] {
+  var values: any[] = [];
```

Comment:
This can be assigned with map using the body of the for loop as the map functions
```
const values = valueSet.map(encodeData);